src/lib/s3Scan.ts

```diff
@@ -0,0 +1,77 @@
+import { Float } from 'aws-sdk/clients/batch'
+import { int } from 'aws-sdk/clients/datapipeline'
+import * as http from 'http'
+import * as Stream from 'stream'
+
+export interface S3ScanPurgeReturn {
+  stream: Stream.Transform
+  deletedCount: int
+  rate: () => Float
+}
+
+export class S3Scan {
+  public static Copy() {
+    //Todo: Implentation Left
+  }
+
+  public static Delete() {
+    //Todo: Implentation Left
+  }
+
+  public static Purge(
+    s3Url: string,
+    options: { [key: string]: string | boolean | http.Agent },
```

Comment:
The options type here is quite complex and it's repeated a few times. Would it make sense to create a type for it with all the possible options in that type?